### ISSUE TITLE
Failed to find semgrep-core in PATH or in the semgrep package.

### ISSUE THREAD

**Issue by aydinnyunus** [semgrep/semgrep#10177]
**Describe the bug**
I use pip for installing semgrep but I am getting this error.

Failed to find semgrep-core in PATH or in the semgrep package.

**To Reproduce**
pip install

**What is the priority of the bug to you?**

- [x] P0: blocking your adoption of Semgrep or workflow
- [ ] P1: important to fix or quite annoying
- [ ] P2: regular bug that should get fixed

**Environment**
i use directly pip install

**Comment by ievans** [comment]
I can't reproduce. Can you share your environmental info? What operating system, version of pip, version of Semgrep, etc.

**Comment by aydinnyunus** [comment]
> I can't reproduce. Can you share your environmental info? What operating system, version of pip, version of Semgrep, etc.

- pip 24.0
- semgrep 1.70.0
- Linux 5.13.0-27-generic #29~20.04.1-Ubuntu

**Comment by muhammadhammad-nc** [comment]
I'm facing the same issue in Kali Linux. before update everything is working as expected but after the Kali Linux update it shows me "Failed to find semgrep-core in PATH or in the semgrep package."

**Comment by tgice** [comment]
Same here, new user, trying to give this a test.  Installed on latest patched Windows 10 Pro, using Python 3.12.6 and Pip 24.3.1.

Have uninstalled (via pip) and re-installed multiple times with no success.

**Comment by nmote** [comment]
The recent reports are likely duplicates of #10652, which was fixed in Semgrep 1.96.0

**Comment by ievans** [comment]
Thanks Nat! I will close, but feel free to reopen if it recurrs after 1.96

**Comment by tgice** [comment]
Hello, I'm using semgrep 1.96 and having the above mentioned error.  I'm a new user, never have installed anything but version 1.96.

Windows 10, latest updates:

> C:\temp>python --version
> Python 3.12.6
> 
> C:\temp>pip --version
> pip 24.3.1 from C:\Python312\Lib\site-packages\pip (python 3.12)
> 
> C:\temp>pip show semgrep
> Name: semgrep
> Version: 1.96.0
> Summary: Lightweight static analysis for many languages. Find bug variants with patterns that look like source code.
> Home-page: https://github.com/returntocorp/semgrep
> Author: Semgrep Inc.
> Author-email: support@semgrep.com
> License:
> Location: C:\Python312\Lib\site-packages
> Requires: attrs, boltons, click, click-option-group, colorama, defusedxml, exceptiongroup, glom, jsonschema, opentelemetry-api, opentelemetry-exporter-otlp-proto-http, opentelemetry-instrumentation-requests, opentelemetry-sdk, packaging, peewee, requests, rich, ruamel.yaml, tomli, typing-extensions, urllib3, wcmatch
> Required-by:
> 
> C:\temp>ver
> 
> Microsoft Windows [Version 10.0.19045.5131]


> C:\temp>dir c:\Python312\Scripts\
>  Volume in drive C has no label.
>  Volume Serial Number is C2C6-DB3F
> 
>  Directory of c:\Python312\Scripts
> 
> 11/11/2024  12:55 PM    <DIR>          .
> 11/11/2024  12:55 PM    <DIR>          ..
> 11/11/2024  12:35 PM           108,382 pip.exe
> 11/11/2024  12:35 PM           108,382 pip3.12.exe
> 11/11/2024  12:35 PM           108,382 pip3.exe
> 11/11/2024  12:55 PM           108,391 pysemgrep.exe
> 11/11/2024  12:55 PM           108,392 semgrep.exe
>                5 File(s)        541,929 bytes

I did originally have Python 3.7.3 installed when I first tried installing semgrep (I think), but it failed in the same way so I upgraded to Python 3.12.6 and tried the install again with pip.

Any other suggestions for something to try?

**Comment by cgdolan** [comment]
Semgrep is currently supported on Windows with through WSL only. Can you please confirm you're running semgrep this way?

Seems like some of our recent prep work for native Windows support has resulted in a confusing "core not found" error when installing without WSL. This wasn't a problem previously since the `semgrep` command simply wasn't put on PATH rendering the installation non-functional.  I've put in a fix for this and it should go out with the next release.

**Comment by tgice** [comment]
Hello, I had *not* been using WSL.  I'll give that a shot.  Thanks.

**Comment by muhammadhammad-nc** [comment]
I had a same problem. So I created a virtual python environment using vpython then install the semgrep inside the vpython environment and it started working. So whenever I need to scan the repo I use vpython environment to access the semgrep.
